app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx

```diff
@@ -203,15 +249,25 @@ export const MentionAutocompleteTextArea = ({
     // Update the text area value
     setTextContent(textWithSelection);
 
-    // Update the cursor to directly after the selection
-    const newCursorPosition = selectionInsertIndex + username.length + 1;
+    // Switch back to the plain text input
+    comboboxRef.current.classList.add('hidden');
+    plainTextAreaRef.current.classList.remove('hidden');
+
+    // Update the cursor to directly after the selection (+2 accounts for the @ sign, and adding a space after the username)
+    const newCursorPosition = selectionInsertIndex + username.length + 2;
     setCursorPosition(newCursorPosition);
   };
 
   useLayoutEffect(() => {
-    if (inputRef.current) {
-      // Replace the whole textarea passed in props with the autocomplete textarea
-      replaceTextArea(replaceElement, inputRef.current);
+    const { current: comboboxTextArea } = comboboxRef;
+    const { current: plainTextArea } = plainTextAreaRef;
+
+    if (comboboxTextArea && plainTextArea) {
```

Comment:
Yeah I thought a little bit about this and ultimately decided we should only introduce it when we need it, otherwise it's extra code/tests we're not using yet (I think we would also need to e.g. check that the tag type is either input or textarea)